namespace.go

```diff
@@ -214,76 +188,39 @@ func (n *Namespace) UpsertPrerendered(
 		totalByteSize += len(chunk)
 	}
 
-	upsertFn := func() error {
-		if resp, err := n.request(
-			ctx,
-			http.MethodPost,
-			"/v1/namespaces/"+n.name,
-			upsertChunks,
-		); err != nil {
-			if resp != nil && resp.Status == http.StatusTooManyRequests {
-				return err // Retryable
-			}
-			return backoff.Permanent(fmt.Errorf("failed to upsert documents: %w", err))
-		}
-		return nil
-	}
-
-	if err := backoff.Retry(upsertFn, backoff.WithContext(backoff.NewExponentialBackOff(), ctx)); err != nil {
-		if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
-			return time.Since(start), totalByteSize, nil
-		}
+	url := fmt.Sprintf("/v1/namespaces/%s", n.ID())
```

Comment:
While here, mind adding a TODO that we should switch this to v2? I have that in a branch somewhere, but it will need some touching up after this PR.

Reply:
Oh yeah, I can just knock that out right now. Done.